### ISSUE TITLE
Can not use Context Menu in tree View

### ISSUE THREAD

**Issue by sunilmittal22** [madcapnmckay/Knockout-UI#4]
Hi Man,

I am exploring your UI Widgts and I really love them . They perform exceptionally well.  But I found some minor issue I think the issue is bit logical. On treeview right now there is no way I can not distinguish between a Right Click and Left Click,. On Right Click on Node I want to show some advanced options and on left click on the Node I want to display details against that Node.

To Made few changes in ui-tree.js to make  it working :

1) In ui-tree.js on Line No 97, I did few changes 
/*Original Script */
            this.selectNode = function () {
                var selected = this.viewModel.selectedNode(), that = this;

```
            if (selected !== undefined && selected.isRenaming()) {
                $('.rename > .node input', viewModel.tree).blur();
            }
            this.saveState();

            if (selected === undefined || (selected !== undefined && selected !== this)) {

                that.viewModel.handlers.selectNode(this, function () {
                    if (selected !== undefined) {
                        selected.isSelected(false);
                        selected.isRenaming(false);
                    }
                    that.isSelected(true);
                    that.viewModel.selectedNode(that);
                    that.saveState();
                });
            }
        }.bind(this);
```

/\* End */

/_Modified Script_/
            this.selectNode = function (event) {
                var selected = this.viewModel.selectedNode(), that = this;

```
            if (selected !== undefined && selected.isRenaming()) {
                $('.rename > .node input', viewModel.tree).blur();
            }
            this.saveState();

            if (selected === undefined || (selected !== undefined && selected !== this)) {

                that.viewModel.handlers.selectNode(event,this, function () {
                    if (selected !== undefined) {
                        selected.isSelected(false);
                        selected.isRenaming(false);
                    }
                    that.isSelected(true);
                    that.viewModel.selectedNode(that);
                    that.saveState();
                });
            }
        }.bind(this);
```

/\* End */

If I am missing something, using which I won't be requiring to change the above. Anyways , I wanted to let you about this.

Thanks Man.. Keep up the Good Work

**Comment by sunilmittal22** [comment]
Hi man..

I am also looking into how we can use the KnockOut UI components as a widget, without interfering with each other.

Like Window manager and Tree View on the same page. I am able to do it but It is not clean . If you can suggest something on this

**Comment by madcapnmckay** [comment]
Hi Sunil,

Apologies for the extremely late reply. I have recently started a new job so have not been very active lately. I reviewed your suggestions and I am unable to consistently pass an event object to the selectNode function as it can be triggered externally and from other methods within the node. Instead I added a right-click handler which you will be able to use as you stated. View the tree example and right click to see the handler fire.

I also created a [jsfiddle](http://jsfiddle.net/madcapnmckay/7eD82/)  for you to see how I would go about using the components together. My other GitHub project uses the components to together extensively so It can be done. Let me know what you think and how this could be improved.

Hope this helps,

Ian
